compute_endpoint/globus_compute_endpoint/cli.py

```diff
@@ -534,6 +539,53 @@ def delete_endpoint(*, name: str, force: bool, yes: bool):
     Endpoint.delete_endpoint(ep_dir, get_config(ep_dir), force=force)
 
 
+@app.command("self-diagnostic")
+@click.option(
+    "-z",
+    "--gzip",
+    "compress",
+    default=False,
+    is_flag=True,
+    help="Save the output to a Gzip-compressed file.",
+)
+@click.option(
+    "--log-size",
+    default=5,
+    help=(
+        "Specify the number of megabytes (MB) to read from log files."
+        " Defaults to 5MB."
+    ),
+)
+@click.help_option("-h", "--help")
+def self_diagnostic(compress: bool, log_size: int):
+    """Run several diagnostic commands to help identify issues.
+
+    This may produce a large amount of output.
+    """
+    log_bytes = log_size * 1024 * 1024
+
+    if not compress:
+        run_self_diagnostic(log_bytes=log_bytes)
+    else:
+        current_date = datetime.now().strftime("%Y-%m-%d")
+        filename = f"globus_compute_diagnostic_{current_date}.txt.gz"
+        temp_filename = f"tmp_{uuid.uuid4()}.txt"
+
+        try:
+            with open(temp_filename, "w") as temp_f:
+                with contextlib.redirect_stdout(temp_f):
+                    run_self_diagnostic(log_bytes=log_bytes)
+
+            with open(temp_filename, "rb") as temp_f:
+                with gzip.open(filename, "wb") as f:
+                    f.writelines(temp_f)
+
+            click.echo(f"Successfully created {filename}")
+        finally:
+            # Ensure we always delete the temp file
+            os.unlink(temp_filename)
```

Comment:
Looks like a good candidate for `tempfile.TemporaryFile`

Reply:
Good thought, but I decided to refactor so we write directly to the `.gz` file.